tests/unit/test_model.py

```diff
@@ -67,29 +72,597 @@ class Person(BaseModel):
     Age=35,
 )
 
+bird_person = Person(
+    "person#7777",
+    "person#birdperson",
+    FirstName="Bird",
+    LastName="Person",
+)
+
+
 @pytest.fixture
-def mock_dynamodb():
+def mock_table_resource():
     mock_resource = mock.Mock()
     mock_resource.return_value = mock_resource
     mock_table = mock.Mock()
     mock_resource.Table.return_value = mock_table
     return mock_resource, mock_table
 
+@pytest.fixture
+def mock_client():
+    mock_client = mock.Mock()
+    mock_client.return_value = mock_client
+    return mock_client
+
 
 @pytest.fixture
 def mock_environ():
+    os.environ['AWS_DEFAULT_REGION'] = 'us-east-1'
     os.environ['AWS_ACCESS_KEY_ID'] = 'testing'
     os.environ['AWS_SECRET_ACCESS_KEY'] = 'testing'
-    os.environ['DYNAMODB_TABLE'] = 'character-table'
+    os.environ['AWS_SECURITY_TOKEN'] = 'testing'
+    os.environ['AWS_SESSION_TOKEN'] = 'testing'
+    os.environ['DYNAMODB_TABLE'] = 'c-132-table'
+
+
+def test_set_defaults():
+    bird_person._set_defaults()
+    bird_person_dict = bird_person.to_dict()
+    assert bird_person_dict['Type'] == 'person'
+    assert bird_person_dict['FirstName'] == 'Bird'
+    assert bird_person_dict['LastName'] == 'Person'
+    assert bird_person_dict['Age'] == 0
+    assert bird_person_dict['PK'] == 'person#7777'
+    assert bird_person_dict['SK'] == 'person#birdperson'
+    assert bird_person_dict['ID'] == '7777'
```

Comment:
This might be a little more readable:
```suggestion
    assert bird_person_dict = {
      'Type': 'person',
    ...
    }
```